### ISSUE TITLE
Error installing on Mac OSX

### ISSUE THREAD

**Issue by mayebejames** [gnosygnu/xowa#687]
Hi, I'm trying to install but am running into this error:

> Unrecognized option: -d64
> Error: Could not create the Java Virtual Machine.
> Error: A fatal exception has occurred. Program will exit

Having found issue #451 I've tried running java -Xmx256m -XstartOnFirstThread -jar xowa_macosx_64.jar   which gets it to open but has some java null errors which force it to close immediately: 

>  arguments:
>   **** NONE ****
>   use --help to show help
> 
> SplashScreen.getSplashScreen() returned null

<img width="1280" alt="Screenshot 2020-03-21 at 07 04 26" src="https://user-images.githubusercontent.com/15050449/77221525-3df98e00-6b42-11ea-8c2f-1a54507a5528.png">

I've tried reinstalling a few times and various parameters but can't seem to get it to work. Sorry if this is obvious but would value some help!

Many thanks,
James

**Comment by gnosygnu** [comment]
Hey, thanks for all the details, as well as the screenshot.

Your steps above are correct. I'm a little puzzled as to what's going on. I downloaded `xowa_app_macosx_64_v4.6.5.1911.zip` and was able to install Simple Wikipedia without an issue. 

When you get a chance, can you get me the following?
* OS X version: Main Menu -> Apple Icon -> About this Mac
  * Mine is `OS X 10.9.5`
* Java version
  * Open Terminal
  * Enter in `java -version`
  * Mine is `1.7.0_80`

**Comment by mayebejames** [comment]
Hey,

Thanks for the quick response!

I'm on Catalina 10.15.3 (19D76)
I downloaded Java today:

java version "14" 2020-03-17

Java(TM) SE Runtime Environment (build 14+36-1461)
Java HotSpot(TM) 64-Bit Server VM (build 14+36-1461, mixed mode, sharing)

Hope you are holding up ok with everything going on at the moment! I've
just got back from working in a hospital in the south pacific with no
internet and realised the importance of an offline wikipedia source! I've
now set up kiwix, but I read that you can use Xowa to creat your own wiki
by creating and editing pages? If so this will be incredibly useful for my
medical knowledge repository!

Best wishes,

James

**Comment by gnosygnu** [comment]
Cool. Thanks for the info.

Can you try downloading / running Java 8? I tried now on my machine and it still works. See steps below.

Thanks!

---

* Go to https://www.java.com
* Click `Download Java` which should bring you to a download page for `Java 8 Update 241`
* Download the file and install the dmg file
* After install, run this to verify the java version
```
"/Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/bin/java" -version
```
* If that reports Java 1.8, then run this to launch xowa
```
"/Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/bin/java" -XstartOnFirstThread -Xmx256m -jar "WHEREVER_YOU_UNZIPPED_XOWA/xowa/xowa/xowa_macosx_64.jar"
```

**Comment by chris-kahn** [comment]
I've got the exact same issue, using Java 1.8, `SplashScreen.getSplashScreen() returned null` and the app just opens to a blank screen and gets frozen with a spinning pinwheel.

**Comment by gnosygnu** [comment]
Hi all. Sorry for the delay. This issue is a bit tough for me to diagnose, because I need to test out changes on a loaner Mac.

I think I have a fix for this issue. Basically, I have to upgrade the SWT from 4.6RC3 to something later. I'll have a new download later this week.

**Comment by gnosygnu** [comment]
I uploaded a fix for the issue. See: https://github.com/gnosygnu/xowa/releases/tag/v4.6.10.2008

It's not a perfect fix, as it looks like the latest Java version + SWT + OSX causes problems. Right now, the main issue is that the XOWA menu is disabled. To enable it, switch to a diferent app and then switch back to XOWA.

I don't know how to fix this, and investigation is hard b/c OSX is not my primary development machine for XOWA. I think the best fix would be to dump SWT entirely and go to Swing. That would be a pretty big task, and is probably not going to be fixed for a while.

Anyway, apologies again for the incomplete nature of the fix as well as the delay.

Technical details below.

---

## What is `java.lang.NullPointerException at org.eclipse.swt.widgets.Control.internal_new_GC`?
This is an SWT + OSX bug. It occurs in version 4.6M3. See: https://bugs.eclipse.org/bugs/show_bug.cgi?id=434393 . Unforturnately, v4.6M3 was the official SWT version for XOWA.

Note that this bug does not occur for Java 7 and earlier. It does seem to surface for Java 8+. I probably forced this to occur when I upgraded XOWA to Java 8 back in April 2020. See #709

This bug was fixed by uploading to the official 4.16 release for 2020-06. See: https://download.eclipse.org/eclipse/downloads/drops4/R-4.16-202006040540/

## What is `Unrecognized option: -d64: Error: Could not create the Java Virtual Machine`?
This is an SWT + OSX + Java bug. 

SWT recommends using `-d64` for OSX. See https://www.eclipse.org/swt/macosx/ and this snippet
```
Launching the application
In the SWTHello example, a small shell script is used as the executable to run. This allows us to easily launch Java with any required options.

#!/bin/sh
BASEDIR=`dirname $0`
exec java \
     -d64 \
     -XstartOnFirstThread \
     -classpath $BASEDIR/swt/swt.jar:$BASEDIR \
     SWTHello
```

However, it looks like JDK started deprecating these obsolete command-line options. See: https://bugs.openjdk.java.net/browse/JDK-4666158

This bug was fixed by removing the `-d64` from `xowa_macosx_64`

## Why doesn't the XOWA menu show or is disabled?
This looks like an SWT + OSX bug. This Arduino link has a fix: https://github.com/arduino/Arduino/issues/6548#issuecomment-332446432

I tried the same by adding the following to the `xowa_macosx_64` shell script: `-Duser.language=en -Duser.country=US -Duser.variant=US -Duser.region=US`

However, this didn't work. After some investigation, the only workaround is to switch to a different app and then switch back to XOWA

Here are some other links for later reference:
* Arduino mentions issue: https://github.com/arx-deidentifier/arx/issues/197
* Recommendation to use `-Dapple.laf.useScreenMenuBar=true` which didn't help: https://stackoverflow.com/questions/8955638/how-do-i-move-my-jmenubar-to-the-screen-menu-bar-on-mac-os-x
* Eclipse ran across this earlier but it affected non-English locales and its fix is very eclipse-specific (Info.plist): https://bugs.eclipse.org/bugs/show_bug.cgi?id=520176
* Another eclipse issue with disabling: https://bugs.eclipse.org/bugs/show_bug.cgi?id=519724
* archimatetool also reports the same issue and Info.plist packaging: https://github.com/archimatetool/archi/issues/304
* Unanswered stackoverflow question with simplest possible code repro: https://stackoverflow.com/questions/25717476/menu-bar-behaving-strangely-on-mac?rq=1

## What is the best way to fix this in the future?
Short-term: Possibly integrate `CocoaUIEnhancer.java` as detailed here: https://eclipsesource.com/blogs/2012/07/19/configuring-the-os-x-application-menu-for-swt-apps/

Long-term: Switch to JavaCEF
* StackOverflow on integrating CEF: https://stackoverflow.com/questions/21192279/how-to-integrate-chromium-embedded-framework-cef-with-java
* Blog post on integrating CEF: https://medium.com/@daniel.bischoff/integrating-chromium-as-a-web-renderer-in-a-java-application-with-jcef-72f67a677db6
* Official link to JavaCEF: https://bitbucket.org/chromiumembedded/java-cef/src/master/
